expr/expression.go

```diff
@@ -33,6 +35,10 @@ func FuncArgFromProto(e *proto.FunctionArgument, baseSchema types.Type, ext Exte
 }
 
 func ExprFromProto(e *proto.Expression, baseSchema types.Type, ext ExtensionLookup, c *extensions.Collection) (Expression, error) {
+	if e == nil {
+		return nil, nil
+	}
```

Comment:
Does this really happen? Not sure as a caller what I'd do if I got back (nil, nil).

Reply:
So I put this for the off-chance where a user passes a `nil` expression or something in the nested protobuf has a `nil` value so that we don't panic with a `dereferenced nil` exception etc. In theory this should only ever happen from either a coding error or a malformed protobuf (I think), but I'd rather avoid the panic if possible by accidentally dereferencing a nil pointer.

Reply:
IMHO, I think raising our own panic here would be preferable, otherwise the caller has to be aware of this case of getting (nil, nil) back (and it's not clear what the right thing to do would be). We could throw our own specific error if we think callers would want to potentially recover from and ignore this situation.

Reply:
Fair point, i'll update this shortly